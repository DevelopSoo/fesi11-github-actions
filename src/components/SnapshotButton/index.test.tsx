import { render } from "@testing-library/react";
import SnapshotButton from ".";

test("Button 컴포넌트 스냅샷 테스트", () => {
  const { container } = render(<SnapshotButton>버튼</SnapshotButton>);
  expect(container).toMatchSnapshot();
});

test("비활성화 상태", () => {
  const { container } = render(<SnapshotButton disabled>클릭</SnapshotButton>);
  expect(container).toMatchSnapshot();
});

test("로딩 상태", () => {
  const { container } = render(<SnapshotButton isLoading>클릭</SnapshotButton>);
  expect(container).toMatchSnapshot();
});
