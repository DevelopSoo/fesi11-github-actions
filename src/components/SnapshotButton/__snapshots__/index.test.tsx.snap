// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Button 컴포넌트 스냅샷 테스트 1`] = `
<div>
  <button
    class="bg-blue-500  "
  >
    버튼
  </button>
</div>
`;

exports[`로딩 상태 1`] = `
<div>
  <button
    class="bg-blue-500  animate-pulse"
  >
    클릭
  </button>
</div>
`;

exports[`비활성화 상태 1`] = `
<div>
  <button
    class="bg-blue-500 cursor-not-allowed opacity-50 "
  >
    클릭
  </button>
</div>
`;
